### PR TITLE
Bridge refactor

### PR DESCRIPTION
**Pull request by wkz**
## Checklist

Tick *relevant* boxes, this PR is-a or has-a:

- [ ] Bugfix
  - [ ] Regression tests
  - [ ] ChangeLog updates (for next release)
- [ ] Feature
  - [ ] YANG model change => revision updated?
  - [ ] Regression tests added?
  - [ ] ChangeLog updates (for next release)
  - [ ] Documentation added?
- [ ] Test changes
  - [ ] Checked in changed Readme.adoc (make test-spec)
  - [ ] Added new test to group Readme.adoc and yaml file
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (please detail in commit messages)
- [ ] Build related changes
- [ ] Documentation content changes
  - [ ] ChangeLog updated (for major changes)
- [ ] Other (please describe):

### CODE REVIEW

File: src/confd/src/ietf-interfaces.c [hunk 1]
```diff
@@ -495,7 +496,7 @@ static int netdag_gen_iface_del(struct dagger *net, struct lyd_node *dif,
 
 	DEBUG_IFACE(dif, "");
 
-	mcast_querier(ifname, 0, 0, 0);
+	/* WKZ TODO mcast_querier(ifname, 0, 0, 0); */
```
Comment (troglobit) [inline src/confd/src/ietf-interfaces.c:499]: This is the only standout issue I can find.

**Review by troglobit — CHANGES_REQUESTED** [review]
Almost like the Texas Chain Saw Massacre ... impressive Christmas cleaning! :exploding_head: 

I only found the one thing, but since the tests pass maybe that commented line can just be dropped now?  I say _maybe_ because with this massive change it is now very hard to see how it all ties together.  Maybe you can draw a picture or update the confd README with some help for us mere mortals?

Another thing, which is more of a *feeling* that I'm struggling to put into words is: the naming of functions in the new files vary and seem excessively long and filled with abbreviations at times.  I think we need some guidelines here.  For example, we could rename infix-if-bridge-port.c to infix-if-brport.c and then simply use brport_ as prefix for all public functions, same for infix-if-bridge-mcd.c -> infix-br-multicast.c (or mcast) then prefix public functions with brmcast_.  All local/private functions do, in my view, not need any prefix.   For example ixif_br_mcd_gen() -> brmcast_gen() and ixif_br_mcd_gen_br_vlan -> gen_br_vlan().

File: src/confd/src/ietf-interfaces.c (hunk 1 above)
Comment (wkz) [reply src/confd/src/ietf-interfaces.c:499]: Thanks! :sweat_smile: 

Yes, this is now taken care of by the `mcd` config generation.

**Review by troglobit — APPROVED** [review]
:shipit: :fire:

**Comment by troglobit** [comment]
@wkz might be a good idea to run the coverity job on this branch as well.  I noticed it failed this passed weekend.  

Just go to https://github.com/kernelkit/infix/actions/workflows/coverity.yml and "Run workflow" on the right-hand side, then select your branch.

**Comment by troglobit** [comment]
Wohoo, syslog support in podman and conmon! 🎉

**Comment by troglobit** [comment]
> > @wkz might be a good idea to run the coverity job on this branch as well. I noticed it failed this passed weekend.
> 
> > 
> 
> > Just go to https://github.com/kernelkit/infix/actions/workflows/coverity.yml and "Run workflow" on the right-hand side, then select your branch.
> 
> 
> 
> @troglobit: For sure, that would be great. The problem with that seems to be that `confd` now depends on a newer `libite` than the one packaged in `ubuntu-latest`. I'm sorry but I've been on sooo many side quests with this PR; I just don't have the energy to fix this at the moment. :tired_face: 

No worries mate, I'll pick up the battle axe next.  It is time for you, our unsung hero, to rest a bit now, finally!

**Pull request merged**